homeassistant/components/cover/__init__.py

```diff
@@ -59,7 +59,7 @@
 
 COVER_SERVICE_SCHEMA = vol.Schema({
     vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
-})
+}, extra=vol.REMOVE_EXTRA)
```

Comment:
We shouldn't do this. This is diverging from how all other services in Home Assistant work. We should throw an error when people submit the wrong data, not silently remove it.

Reply:
I would agree if the intend would have been to catch invalid data, but here it's just to enable greater functionality for `service_template`. IMO it could be quite useful to decide, like here, based on values from a script or automation which service to call, however because `set_cover_position` needs a `position` value I can't combine all services into one `service_template`. Which of the options to choose [1] ALLOW_EXTRA, [2] REMOVE_EXTRA or even [3] add the optional position parameter to all other cover services basically dosn't matter. As a side note: I would even go as far as to suggest to add one of the above to the gerneral service schema.
However you decide I will not open any other PR's regarding this. I would really like if this could be added to the repo but if it isn't I will manually change it for my HA instance after every update.

Reply:
If `service_template` is too limited, consider using the `python_script` component or a custom component that registers your own service.

Home Assistant data validation fails hard and that's a good thing. Too much users would be confused why their things won't work otherwise.